src/widgets/Modal/ModalHeader.tsx

```diff
@@ -1,16 +1,22 @@
-import React from "react";
+import React, { useContext } from "react";
 import { Flex } from "theme-ui";
 import style from "./styles";
 import { IconButton } from "../../components/Button";
 import { InternalProps } from "./types";
 import { Divider } from "../../components/Divider";
+import { Context as ModalContext } from "./ModalContext";
 
-const ModalHeader: React.FC<InternalProps> = ({ children, handleClose, ...props }) => {
+const ModalHeader: React.FC<InternalProps> = ({ children, onDismiss, ...props }) => {
+  const { handleClose } = useContext(ModalContext);
+  const onClose = () => {
+    onDismiss?.();
+    handleClose();
+  };
```

Comment:
Can we just keep onDismiss and pass that? It would make these components cleaner

Reply:
This was a request from one of the consultants. He mentioned he needed something like this up so he could apply it to something he was working on, and that it was harder to get a fix without having this.